Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -507,6 +505,7 @@ class TIFFLoadingContext {
             next_tag_offset += 12;
         }
 
+        m_ifd_stack.take_last();
```

Comment:
Why do we pop down here instead or right after accessing `last()` at the start of the function?

Also, in the next patch we do

```
            TRY(push_ifd(ifd_offset));
            TRY(read_next_image_file_directory());
```

back to back -- why do we need a stack for these at all then?

Sorry, I have a feeling I'm thoroughly misunderstanding things :yaksad:

Reply:
That's on me, a stack is not necessary at all (to store the IFD). I implemented a stack because that's how I understood the traversal is meant to be done. But it's exactly what the call stack is doing 😅.
Thanks for catching that!